service-base/src/main/java/org/eclipse/hono/service/credentials/CredentialsService.java

```diff
@@ -45,21 +45,47 @@
+     * @throws NullPointerException if any of the parameters is {@code null}.
      */
-    void addCredentials(String tenantId, String deviceId, String type, String authId, JsonObject otherKeys, Handler<AsyncResult<CredentialsResult>> resultHandler);
+    void addCredentials(String tenantId, String deviceId, String type, String authId, JsonObject credentialsObject, Handler<AsyncResult<CredentialsResult>> resultHandler);
 
     /**
-     * Updates device registration data.
-     * TODO: params
+     * Updates device credentials data.
+     * @param tenantId The tenant the device belongs to.
+     * @param deviceId The ID under which the device is registered.
+     * @param type The type of credentials to update.
+     * @param authId The authID of the device to update credentials data for.
+     * @param credentialsObject A map containing keys and values that fulfill the credentials format of the credentials api.
+     *                  See <a href="https://www.eclipse.org/hono/api/Credentials-API/#credentials-format">Credentials Format</a> for details.
+     * @param resultHandler The handler to invoke with the result of the operation. If a device with the given ID does not
+     *         exist for the tenant, or there are no credentials currently registered for this device that match the type and authId,
+     *                     the <em>status</em> will be {@link HttpURLConnection#HTTP_NOT_FOUND}.
+     *         Otherwise the operation is successful and the status will be {@link HttpURLConnection#HTTP_NO_CONTENT}.
+     * @throws NullPointerException if any of the parameters is {@code null}.
      */
-    void updateCredentials(String tenantId, String deviceId, String type, String authId, JsonObject otherKeys, Handler<AsyncResult<CredentialsResult>> resultHandler);
+    void updateCredentials(String tenantId, String deviceId, String type, String authId, JsonObject credentialsObject, Handler<AsyncResult<CredentialsResult>> resultHandler);
 
     /**
-     * Removes a device.
-     * TODO: params
+     * Removes device credentials data.
+     * @param tenantId The tenant the device belongs to.
+     * @param deviceId The ID under which the device is registered.
+     * @param type The type of credentials to remove.
+     * @param authId The authID of the device to remove credentials data for.
+     * @param resultHandler The handler to invoke with the result of the operation. If a device with the given ID does not
+     *         exist for the tenant, or there are no credentials currently registered for this device that match the type and authId,
+     *                     the <em>status</em> will be {@link HttpURLConnection#HTTP_NOT_FOUND}.
+     *         Otherwise the operation is successful and the status will be {@link HttpURLConnection#HTTP_NO_CONTENT}.
+     * @throws NullPointerException if any of the parameters is {@code null}.
```

Comment:
Correct, and the '*' as special value for type was missing as well - will fix that too.